routes/api/v1/signup.js

```diff
@@ -0,0 +1,30 @@
+var express = require('express');
+var router = express.Router();
+
+router.post('/', (req, res, next) => {
+  var db = req.app.get('db')
+  var user = req.body
+
+  if(user.username && user.password) {
+    db('users').where({username: user.username})
+    .then( (existingUser) => {
+      if(existingUser[0]) {
+        return res.json({errorMessage: 'Username already exists'})
+      } else {
+        //hash password with bcrypt before saving
+        db('users').insert(user)
+        .then( () => {
+          //generate JWT
+          //return JWT
+          return res.json()
+        });
+      }
+    });
+    }else{
+      var err = new Error('Not found');
+      err.status = 400;
+      next(err);
+    }
```

Comment:
Yes, it is basically what you described in that you definitely want a `lib` folder and inside there, you'd also have a `model` or `models` folder. I've seen it both ways with and without `models` being nested within `lib`. It's really a personal preference as there is no real right or wrong there as it relates to the resulting functionality.

I have gone back and forth myself about the most optimal and obvious directory structure and I've not yet settled on anything that works for every project. For some projects one structure seems to fit a little more than other approaches. 

At the end of the day, the rails app folder structure is fine and if that feels natural, go with it and you can change it down the line if you really feel strongly for some other pattern.